### ISSUE TITLE
[java] DataflowAnomalyAnalysis DU false positive

### ISSUE THREAD

**Issue by Reissner** [pmd/pmd#399]
Rule: [DataflowAnomalyAnalysis](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#dataflowanomalyanalysis)

the following yields false DU positive on variable `cand`: 

Code:
``` java
public class Test {
    public int indexOf(Object obj) {
	// throws ClassCastException 
	Integer dig = (Integer)obj;
	if (dig != 0 && dig != 1) {
	    return -1;
	}
	// throws IllegalArgumentException and NullPointerException 
	boolean cand = int2bool(dig);//<cand
	for(int i = 0; i < this.size; i++) {
	    if (cand == this.digits[i]) {
		return i;
	    }
	}
	return -1;
    }
}
```

Result:
`n/a:9:	Found 'DU'-anomaly for variable 'cand' (lines '9'-'16').`

**Comment by adangel** [comment]
The rule [DataflowAnomalyAnalysis](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#dataflowanomalyanalysis) has been deprecated in favour of [UnusedAssignment](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedassignment), which is available with PMD 6.26.0.

Therefore this issue wont be fixed.

See also #2647.
